control-plane/config-entries/controllersv2/mesh_gateway_controller.go

```diff
@@ -261,3 +283,40 @@ func (r *MeshGatewayController) getGatewayClassForGateway(ctx context.Context, g
 	}
 	return &gatewayClass, nil
 }
+
+func areServicesEqual(a, b *corev1.Service) bool {
+
+	if a == nil || b == nil {
```

Comment:
Should this be `&&`? Or do you accept them as equal if either one is `nil`?

Reply:
If either one of them is nil, you don't want to try and merge the nil service, so we consider them to be equal.

Reply:
Gotcha! Maybe there could be a comment above the line to clarify.